src/rpc/handlers/LedgerEntry.cpp

```diff
@@ -97,11 +103,29 @@ LedgerEntryHandler::process(LedgerEntryHandler::Input input, Context const& ctx)
         auto const owner = util::parseBase58Wrapper<ripple::AccountID>(
             boost::json::value_to<std::string>(input.depositPreauth->at(JS(owner)))
         );
-        auto const authorized = util::parseBase58Wrapper<ripple::AccountID>(
-            boost::json::value_to<std::string>(input.depositPreauth->at(JS(authorized)))
-        );
+        // Only one of authorize or authorize_credentials MUST exist;
+        if (!input.depositPreauth->contains(JS(authorized)) &&
+            !input.depositPreauth->contains(JS(authorize_credentials)))
+            return Error{Status{RippledError::rpcBAD_CREDENTIALS, "malformedAuthorizeCredentials"}};
+
+        if (input.depositPreauth->contains(JS(authorized))) {
+            if (input.depositPreauth->contains(JS(authorize_credentials)))
+                return Error{Status{RippledError::rpcBAD_CREDENTIALS, "malformedAuthorizeCredentials"}};
 
-        key = ripple::keylet::depositPreauth(*owner, *authorized).key;
+            auto const authorized = util::parseBase58Wrapper<ripple::AccountID>(
+                boost::json::value_to<std::string>(input.depositPreauth->at(JS(authorized)))
+            );
+            key = ripple::keylet::depositPreauth(*owner, *authorized).key;
+        } else {
+            auto const authorizedCredentials =
+                util::parseAuthorizeCredentials(input.depositPreauth->at(JS(authorize_credentials)).as_array());
+            // TODO: Move this check into validation.hpp
+            if (authorizedCredentials.size() > ripple::maxCredentialsArraySize)
```

Comment:
Do we need to check the emptiness of `authorizedCredentials`?

Reply:
checked